Source/sha.cpp

```diff
@@ -1,7 +1,40 @@
 #include "diablo.h"
 
+#include <cstdint>
+
 DEVILUTION_BEGIN_NAMESPACE
 
+// Diablo's "SHA1" is different from actual SHA1 in that it uses arithmetic
+// right shifts (sign bit extension).
+//
+// Here we check whether the platform does signed-bit extension.
+// If not, we fall back to a portable implementation.

```

Comment:
Why not simply always use the portable code?

Reply:
According to this StackOverflow anser, older compilers are not as efficient with the portable version https://stackoverflow.com/a/53766752/181228

Reply:
Since this only truggeres during save/load I would rather that the code is tested havily then that it performce fast on all systems.

Reply:
Done